homeassistant/components/netatmo/sensor.py

```diff
@@ -57,175 +58,202 @@
     "guststrength",
     "sum_rain_1",
     "sum_rain_24",
-]
+)
+
 
+@dataclass
+class NetatmoSensorEntityDescription(SensorEntityDescription):
+    """Describes Netatmo sensor entity."""
 
-class SensorMetadata(NamedTuple):
-    """Metadata for an individual sensor."""
+    _netatmo_name: str | None = None
 
-    name: str
-    netatmo_name: str
-    enable_default: bool
-    unit: str | None = None
-    icon: str | None = None
-    device_class: str | None = None
+    def __post_init__(self) -> None:
+        """Ensure all required attributes are set."""
+        if self._netatmo_name is None:
+            # pragma: no cover
+            raise TypeError
+        self.netatmo_name = self._netatmo_name
```

Comment:
I truly wonder if we should do this or not. This is a runtime check for a static list.

Reply:
True, it is a runtime check. However, the performance impact should be limited as it's only executed during when the file is loaded, i.e. for integration setup.

The alternative would probably be something like
```py
    self.netatmo_name = cast(str, self._netatmo_name)
```
That doesn't catch a missing `_netatmo_name` though.

Reply:
Alternatively, it could add an property method to the dataclass that returns the string and it typed as such.

```py
@dataclass
class NetatmoSensorEntityDescription(SensorEntityDescription):
    """Describes Netatmo sensor entity."""

    _netatmo_name: str | None = None

    @property
    def netatmo_name(self) -> str:
        if self._netatmo_name is None:
            raise AttributeError
        return _netatmo_name
     
	@name.setter
    def netatmo_name(self, name: str) -> None:
        self._netatmo_name = name
```

It is a bit more explicit maybe.

Reply:
Just my opinion, but I still prefer the current solution. Since `None` is only used as default value, I would want my code to fail early and loud if I missed a required field.